util/calc.py

```diff
@@ -191,6 +190,26 @@ def build_func_map(parameter, chunk, coefficients):
     return args
 
 
+def in_range(start, stop, time):
+    return (start <= time 
+        and (stop > time 
+            or (stop == time and stop == start)
+            or stop == None))
+
+
+def to_numpy(value):
+    return numpy.array(value) if (type(value) == list) else value
+
+
+def build_CC_argument(frames, times):
+    r = []
+    for t in times:
+        f = next((f for f in frames if (in_range(f.get('start'), f.get('stop'), t))), None)
+        if (f != None):
+            r.append(to_numpy(f.get('value')))
+    return r
+
+
```

Comment:
Need to refactor this logic to use numpy, as we may be manipulating hundreds of thousands of records. I think something like this:

```
def build_CC_argument(frames, times):
    cc = numpy.empty(times.shape)
    cc[:] = numpy.NAN
    frames = [(f['start'], f['stop'], f['value']) for f in frames]
    frames.sort()

    for frame in frames[::-1]:
        mask = numpy.logical_and(times >= frame[0], times < frame[1])
        cc[mask] = frame[2]
    return cc
```

The sorting and reversing is to keep your logic that prefers the CC with the earlier start time. If we actually prefer the later start time then we shouldn't reverse the list when iterating.